src/main/java/ezschedule/ui/Calendar.java

```diff
@@ -0,0 +1,169 @@
+        year.setText(String.valueOf(date.getYear()));
+        month.setText(String.valueOf(date.getMonth()));
+
+        // List of activities for a given month
+        Map<Integer, List<Event>> eventsForMonthMap = getEventsForMonth();
+
+        // Check for leap year
+        if (isLeapYear()) {
+            monthMaxDate = 28;
+        }
+        drawCalenderBoxes(eventsForMonthMap);
+    }
+
+    private Map<Integer, List<Event>> getEventsForMonth() {
+        List<Event> events = new ArrayList<>();
+        eventList.forEach(e -> {
+            if (isInMonth(e)) {
+                events.add(e);
+            }
+        });
+        return getCalenderMap(events);
+    }
+
+    private Map<Integer, List<Event>> getCalenderMap(List<Event> events) {
+        Map<Integer, List<Event>> calendarEventsMap = new HashMap<>();
+        for (Event e : events) {
+            int date = e.getDate().getDay();
+            if (!calendarEventsMap.containsKey(date)) {
+                calendarEventsMap.put(date, List.of(e));
+            } else {
+                List<Event> oldList = calendarEventsMap.get(date);
+                List<Event> newList = new ArrayList<>(oldList);
+                newList.add(e);
+                calendarEventsMap.put(date, newList);
+            }
+        }
+        return calendarEventsMap;
+    }
+
+    private void drawCalenderBoxes(Map<Integer, List<Event>> eventsForMonthMap) {
+        int dateOffset = ZonedDateTime.of(date.getYear(), date.getMonthValue(), 1, 0,
+                0, 0, 0, date.getZone()).getDayOfWeek().getValue() % 7;
+
+        for (int i = 0; i < 6; i++) {
+            for (int j = 0; j < 7; j++) {
+                int calculatedDate = (j + 1) + (7 * i);
+                if (calculatedDate > dateOffset) {
+                    int currentDate = calculatedDate - dateOffset;
+                    if (currentDate <= monthMaxDate) {
+                        String date = String.valueOf(currentDate);
+                        List<Event> eventsForCurrentDate = eventsForMonthMap.get(currentDate);
+                        calendar.getChildren().add(new CalendarBox(
+                                isToday(currentDate), date, eventsForCurrentDate, filterExecutor).getRoot());
+                    }
+                } else {
+                    calendar.getChildren().add(new CalendarBox().getRoot());
+                }
+            }
+        }
+    }
+
+    private boolean isInMonth(Event event) {
+        int year = date.getYear();
+        int month = date.getMonth().getValue();
+        return event.getDate().getYear() == year && event.getDate().getMonth() == month;
+    }
+
+    private boolean isToday(int currentDate) {
+        return today.getYear() == date.getYear()
+                && today.getMonth() == date.getMonth()
+                && today.getDayOfMonth() == currentDate;
+    }
+
+    private boolean isLeapYear() {
+        return date.getYear() % 4 != 0 && monthMaxDate == 29;
```

Comment:
Shouldn't (leap years % 4 == 0) if it is a leap year?

Also,
> There is a leap year every year whose number is perfectly divisible by four - except for years which are both divisible by 100 and not divisible by 400. The second part of the rule effects century years. For example; the century years 1600 and 2000 are leap years, but the century years 1700, 1800, and 1900 are not.

Adding an additional condition `if (isCenturyYear) {date.getYear() % 400 ==0;}` may prevent this loophole.

Reply:
Resolved. The method should be name isNotLeapYear()